src/app/Providers/HostsPage.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import {
+  PageSection,
+  Level,
+  LevelItem,
+  Bullseye,
+  Card,
+  CardBody,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  Spinner,
+  Title,
+} from '@patternfly/react-core';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+import { Breadcrumb, BreadcrumbItem } from '@patternfly/react-core';
+import { Link, useRouteMatch } from 'react-router-dom';
+import VMwareProviderHostsTable from './components/VMwareProviderHostsTable';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+
+export interface IHostsMatchParams {
+  url: string;
+  providerId: string;
+}
+
+export const HostsPage: React.FunctionComponent = () => {
+  const match = useRouteMatch<IHostsMatchParams>({
+    path: '/providers/:providerId',
+    strict: true,
+    sensitive: true,
+  });
+
+  const isFetchingInitialHosts = false;
+
+  return (
+    <>
+      <PageSection variant="light">
+        <Level>
+          <LevelItem>
+            <Breadcrumb>
+              <BreadcrumbItem>
+                <Link to={`/providers`}>Providers</Link>
+              </BreadcrumbItem>
+              <BreadcrumbItem isActive>{match?.params.providerId} - hosts</BreadcrumbItem>
+            </Breadcrumb>
+          </LevelItem>
+        </Level>
+        <Level className={spacing.mtLg}>
+          <LevelItem>
+            <Title headingLevel="h1">Hosts - {match?.params.providerId}</Title>
+          </LevelItem>
+        </Level>
+      </PageSection>
+      <PageSection>
+        {isFetchingInitialHosts ? (
+          <Bullseye>
+            <EmptyState>
+              <div className="pf-c-empty-state__icon">
+                <Spinner size="xl" />
+              </div>
+              <Title headingLevel="h2">Loading...</Title>
+            </EmptyState>
+          </Bullseye>
+        ) : (
+          <Card>
+            <CardBody>
+              {!match?.params.providerId ? (
+                <EmptyState className={spacing.my_2xl}>
+                  <EmptyStateIcon icon={PlusCircleIcon} />
+                  <Title headingLevel="h2" size="lg">
+                    No hosts
+                  </Title>
+                  <EmptyStateBody>No hosts available for this provider.</EmptyStateBody>
+                </EmptyState>
```

Comment:
This "no hosts" message only appears if the route doesn't have a providerId, but not if the hosts are loaded and that list is empty. We should probably handle that case, and maybe even just have the `providerId` prop of VMwareProviderHostsTable be nullable/optional, pass in whatever we've got here, and have a condition in there that will show "no hosts" if either the `providerId` is undefined or if the hosts array is empty.